### PR TITLE
Error reporting quickstart

### PR DESCRIPTION
**Pull request by jabubake**

### CODE REVIEW

File: errorreporting/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-errorreporting</artifactId>
+      <version>0.24.0-alpha</version>
+    </dependency>
+    <!-- Test dependencies -->
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
```
Comment (lesv) [inline errorreporting/pom.xml]: `<version>`

File: errorreporting/src/main/java/com/example/errorreporting/QuickStart.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.google.devtools.clouderrorreporting.v1beta1.ErrorContext;
+import com.google.devtools.clouderrorreporting.v1beta1.ProjectName;
+import com.google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent;
+import com.google.devtools.clouderrorreporting.v1beta1.SourceLocation;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+/**
+ * Snippet demonstrates using the Google Cloud Error Reporting API to report a custom error event.
+ */
```
Comment (lesv) [inline errorreporting/src/main/java/com/example/errorreporting/QuickStart.java]: Most users won't be using this kind of code, please see StackDriver Logging w/ link.

File: flexible/errorreporting/src/main/java/com/example/flexible/errorreporting/ErrorReportingExample.java [hunk 3]
```diff
@@ -53,13 +53,12 @@ private void logCustomErrorEvent() {
     try (ReportErrorsServiceClient reportErrorsServiceClient = ReportErrorsServiceClient.create()) {
       // custom exception logged using the API
       Exception e = new Exception("custom event reported using the API");
-      // Events reported using the API must contain a stack trace message
-      StringWriter stackTrace = new StringWriter();
-      e.printStackTrace(new PrintWriter(stackTrace));
+      StringWriter sw = new StringWriter();
+      e.printStackTrace(new PrintWriter(sw));
       ReportedErrorEvent errorEvent =
```
Comment (lesv) [inline flexible/errorreporting/src/main/java/com/example/flexible/errorreporting/ErrorReportingExample.java]: Just weird. 56-57 & L61

Comment (lesv) [reply flexible/errorreporting/src/main/java/com/example/flexible/errorreporting/ErrorReportingExample.java]: If we are going to do this, we really should explain what's happening.

e.printStackTrace(PrintWriter(StringWriter))
.setMessage(StringWriter.toString())

File: errorreporting/pom.xml (hunk 1 above)
Comment (jabubake) [reply errorreporting/pom.xml]: Done

File: errorreporting/src/main/java/com/example/errorreporting/QuickStart.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import com.google.cloud.errorreporting.v1beta1.ReportErrorsServiceClient;
+import com.google.devtools.clouderrorreporting.v1beta1.ErrorContext;
+import com.google.devtools.clouderrorreporting.v1beta1.ProjectName;
+import com.google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent;
+import com.google.devtools.clouderrorreporting.v1beta1.SourceLocation;
+
+/**
+ * Snippet demonstrates using the Stackdriver Error Reporting API to report a custom error event.
+
+ * This library is not required on App Engine, errors written to stderr are automatically written
```
Comment (jabubake) [inline errorreporting/src/main/java/com/example/errorreporting/QuickStart.java]: @lesv PTAL if this suffices, I removed the stack trace reporting as well to make it clearer that using this API is an extreme rare case.

**Pull request merged**